From f61a3919b316475b256acda9cd742dfc84b617cc Mon Sep 17 00:00:00 2001
From: Lars Uebernickel <lars.uebernickel@canonical.com>
Date: Thu, 11 Dec 2014 17:49:02 +0100
Subject: [PATCH] Menu items: force loading 16x16 icons

GtkIconTheme doesn't scale icons beyond the size specified in the theme
anymore. This can result in arbitrarily large menu items when a theme
only provides large icons.

Force icons to always be 16x16 to ensure that menu items always have the
same height.

https://bugzilla.gnome.org/show_bug.cgi?id=741259
---
 gtk/deprecated/gtkimagemenuitem.c | 1 +
 gtk/gtkmodelmenuitem.c            | 1 +
 2 files changed, 2 insertions(+)

diff --git a/gtk/deprecated/gtkimagemenuitem.c b/gtk/deprecated/gtkimagemenuitem.c
index e308d51..1b46ead 100644
--- a/gtk/deprecated/gtkimagemenuitem.c
+++ b/gtk/deprecated/gtkimagemenuitem.c
@@ -1126,6 +1126,7 @@ gtk_image_menu_item_set_image (GtkImageMenuItem *image_menu_item,
                 "visible", show_image (image_menu_item),
                 "no-show-all", TRUE,
                 NULL);
+  gtk_image_set_pixel_size (GTK_IMAGE (image), 16);
 
   g_object_notify (G_OBJECT (image_menu_item), "image");
 }
diff --git a/gtk/gtkmodelmenuitem.c b/gtk/gtkmodelmenuitem.c
index dff3503..8591a7e 100644
--- a/gtk/gtkmodelmenuitem.c
+++ b/gtk/gtkmodelmenuitem.c
@@ -199,6 +199,7 @@ gtk_model_menu_item_set_icon (GtkModelMenuItem *item,
       GtkWidget *image;
 
       image = gtk_image_new_from_gicon (icon, GTK_ICON_SIZE_MENU);
+      gtk_image_set_pixel_size (GTK_IMAGE (image), 16);
       gtk_box_pack_start (GTK_BOX (child), image, FALSE, FALSE, 0);
       gtk_widget_show (image);
     }
-- 
2.1.3

